### ISSUE TITLE
Configuring QGIS-Server URL (? and &)

### ISSUE THREAD

**Issue by Xavier-github** [qgis/qwc2-demo-app#158]
Hi all,

I just tested qwc2, it looks promising. But I have a difficulty in setting up my own maps, through qgis-web-server. Indeed:
 * In my case, QGIS Server works like described in https://issues.qgis.org/projects/qgis/wiki/QGIS_Server_Tutorial#Map-parameter : that is I set up the URL with http://myserver.com/cgi/qgis_mapserv.fcgi?MAP=/home/user/mymap.qgs
 * I can "yarn start" and get the thumbnail corresponding to "mymap.qgs" (with colors corresponding to my layers).
 * When I open it, the map is loading without the layers. When I check the layers with the menu on the right, they appear to be selected with a tick.
 * I state that, at this place, images are missing. The explanation is that their URL look like http://myserver.com/cgi/qgis_mapserv.fcgi?MAP=/home/user/mymap.qgs?SERVICE=WMS&VERSION=...
--> I should avoid the two "?", the second one should already be "&" (I checked that such an URL works). But I do not know how to do that... and if it would correct at the same time the fact that the layers do not appear on the map.

Does anybody know how to do that? I stated that there were no "?" in the URL given at example, it my be a solution too - but I do not know either how to do that.
Thanks in advance,
Xavier

**Comment by manisandro** [comment]
Can you post your themesConfig.json and the resulting themes.json?

**Comment by Xavier-github** [comment]
Yes for sure! Please find them attached.

[themes.json.txt](https://github.com/qgis/qwc2-demo-app/files/3580416/themes.json.txt)
[themesConfig.json.txt](https://github.com/qgis/qwc2-demo-app/files/3580417/themesConfig.json.txt)

**Comment by manisandro** [comment]
I don't see the double `?` issue, but there was a case sensitivity issue in QWC: only the lowecase `map`  query-parameter was handled correctly, not the upper-case `MAP`. Should be fixed in https://github.com/qgis/qwc2/commit/813a15a77a1a4f3d911732fa8f38b14c2c833a51

**Comment by Xavier-github** [comment]
I downloaded the file but do not know which file to replace. The one in QWC2Components/utils/ does not look like this one, and I did not find any other LayerUtils.js?

May it be that I downloaded the files (in Github) through qgis/qwc2-demo-app, choosing "stable version"... that is by using "git clone --recursive https://github.com/qgis/qwc2-demo-app.git" before "cd qwc2-demo-app" and "yarn install"?

**Comment by manisandro** [comment]
You shouldn't be updating files manually, let git do it for you. Either start anew

    git clone --recursive https://github.com/qgis/qwc2-demo-app.git

or update your existing clone, see https://github.com/qgis/qwc2-demo-app/blob/master/doc/QWC2_Documentation.md#keeping-the-qwc2-application-up-to-date

**Comment by Xavier-github** [comment]
The solution of updating my clone seems quite interesting, in order not to lose my config files (even if I can easily modify it again). That's why I'm interested with the second solution, but I do not understand what `<tag>` refers to (I tried `stable` and `<stable>`, it does not seem to work). Could you tell me what I should write instead of `<tag>`?
`cd qwc2`
`git fetch`
`git checkout <tag>`

**Comment by manisandro** [comment]
You can see what tags are available here

![image](https://user-images.githubusercontent.com/1298852/64380374-1ec96200-d031-11e9-976b-ec8c64ac08ce.png)

in this case you'll however want to checkout `master`, as the commit indicated above is not yet part of any tagged version.

**Comment by Xavier-github** [comment]
Thank you. That was done with:
```
git fetch
git pull
```
and checked with:
```
$ git checkout master
Déjà sur 'master'
Votre branche est à jour avec 'origin/master'.
```

But now, I've got now 1 error starting with "ERROR in ./js/app.jsx" (1) and 16 with "ERROR in ./js/appConfig.js"; and when I try to connect to localhost:8081, nothing appears and the browser console delivers "La ressource à l’adresse « http://localhost:8081/dist/QWC2App.js » a été bloquée en raison d’un type MIME (« text/html ») incorrect (X-Content-Type-Options: nosniff)"

When I "yarn start", I noticed following lines. May it be that the origin is the "webpack output"? I have no idea what these words refer to.
```
Project is running at http://localhost:8081/
webpack output is served from /dist/
Content not from webpack is served from ./
```

Sincerely,
Xavier

**Comment by manisandro** [comment]
Not sure what happened, but unless one of the documented incompatible changes (see [1]) applies and you need to update some files accordingly, I'd just start anew by cloning the demo app again.

[1] https://github.com/qgis/qwc2-demo-app/blob/master/ChangeLog.md

**Comment by Xavier-github** [comment]
Thank you for your help. I do not know either what was happening, but it seems that it was due to this "stable" version instead of "master". I could not find an incompatible change in my case: but I started anew the installation of the master branch, as described in [https://github.com/qgis/qwc2-demo-app/blob/master/doc/QWC2_Documentation.md#quick-start](https://github.com/qgis/qwc2-demo-app/blob/master/doc/QWC2_Documentation.md#quick-start).
All is working now, my map is online, and I even see the images with the legends of my layers.

--> Thanks again, I close this subject and summarize this way: I just had to leave the "stable" version and to install the "master" one _from the beginning_.
